R/modules.R

```diff
@@ -108,34 +108,42 @@ append_module <- function(modules, module) {
 }
 
 #' Does the object make use of `teal.reporter` reporting
-#' @param modules `teal_module` or `teal_modules` object
+#' @param modules (`teal_module` or `teal_modules`) object
+#' @param args (`character`) names of the arguments to be checked against formals of `teal` modules.
 #' @return `logical` whether the object makes use of `teal.reporter` reporting
-#' @rdname is_reporter_used
+#' @rdname is_args_used
 #' @keywords internal
-is_reporter_used <- function(modules) {
-  UseMethod("is_reporter_used", modules)
+is_args_used <- function(modules, args) {
+  UseMethod("is_args_used", modules)
 }
 
-#' @rdname is_reporter_used
+#' @rdname is_args_used
 #' @keywords internal
-is_reporter_used.default <- function(modules) {
-  stop("is_reporter_used function not implemented for this object")
+#' @export
+is_args_used.default <- function(modules, args) {
+  stop("is_args_used function not implemented for this object")
 }
 
-#' @rdname is_reporter_used
+#' @rdname is_args_used
 #' @export
 #' @keywords internal
```

Comment:
do we need to export this?

Reply:
It's not exported formally. @export tag over S3 method registers this method in the namespace (needed to dispatch by the type). If we don't put @export here the method `is_args_used.teal_module` will be treated as is instead of being `is_args_used` dispatched by type.

```
S3method(is_args_used,teal_module)
```